Modules/selectmodule.c

```diff
@@ -702,39 +748,6 @@ poll_dealloc(pollObject *self)
     PyObject_Del(self);
 }
 
-static PyTypeObject poll_Type = {
```

Comment:
Would it be possible to not move this code, to group all code related to poll at the same place, as done currently.

Reply:
Currently we must either move all method and type definitions towards the end of the file, or move all typedefs and type declarations to the beginning. This is because `#include "clinic/selectmodule.c.h" must come between them.

On [bpo-20180](https://bugs.python.org/issue20180) ([msg304976](https://bugs.python.org/issue20180#msg304976)), regarding `itertools`, @serhiy-storchaka mentioned preference for moving the method and type definitions to the end rather than the alternative. Therefore I did the same here.

Reply:
This is not a preference, this is just an option. Since methods lists and type object initializers could be generated by Argument Clinic in future, I think it is easier to move them.